Source/ExecutionEngine/ExecutionEngine.cs

```diff
@@ -660,6 +654,14 @@ public void Inline(Program program)
       }
     }
 
+    [Obsolete("Please inline this method call")]
+    public PipelineOutcome InferAndVerify(
+      Program program,
+      PipelineStatistics stats,
+      string programId = null,
+      ErrorReporterDelegate er = null, string requestId = null) {
+      return InferAndVerify(Console.Out, program, stats, programId, er, requestId).Result;
+    }
```

Comment:
Feels weird to have two overloads with one of them `async` and the other one not..

Reply:
One of them is only there for making migration easier. Does that help? Do you want me to remove the non-async one?

Reply:
I think it's fine for now to have the two versions,

Reply:
> Having both is fine. I expected the async one to be called InferAndVerifyAsync

I've seen that in some places, but putting part of the method signature in the name seems like a duplication of information to me. I'm guessing the main purpose of having Async as a name suffix might be for exactly this scenario, where there's an old and a new API. What do you think?

Reply:
I think it's fine, no need to worry about it too much.